sdk/servicebus/Microsoft.Azure.ServiceBus/src/MessageReceivePump.cs

```diff
@@ -63,23 +66,24 @@ async Task MessagePumpTaskAsync()
 
                     TaskExtensionHelper.Schedule(async () =>
                     {
-                        Message message = null;
+                        IList<Message> messages = null;
+
                         try
                         {
-                            message = await this.messageReceiver.ReceiveAsync(this.registerHandlerOptions.ReceiveTimeOut).ConfigureAwait(false);
-                            if (message != null)
+                            messages = await this.messageReceiver.ReceiveAsync(this.registerHandlerOptions.MaxBatchSize, this.registerHandlerOptions.ReceiveTimeOut).ConfigureAwait(false);
+                            if (messages != null)
                             {
-                                MessagingEventSource.Log.MessageReceiverPumpTaskStart(this.messageReceiver.ClientId, message, this.maxConcurrentCallsSemaphoreSlim.CurrentCount);
+                                MessagingEventSource.Log.MessageReceiverPumpTaskStart(this.messageReceiver.ClientId, messages, this.maxConcurrentCallsSemaphoreSlim.CurrentCount);
 
                                 TaskExtensionHelper.Schedule(() =>
                                 {
                                     if (ServiceBusDiagnosticSource.IsEnabled())
                                     {
-                                        return this.MessageDispatchTaskInstrumented(message);
+                                        return this.MessageDispatchTaskInstrumented(messages, this.registerHandlerOptions.MaxBatchSize);
```

Comment:
Just have `MessageDispatchTaskInstrumented` use `this.registerHandlerOptions.MaxBatchSize`